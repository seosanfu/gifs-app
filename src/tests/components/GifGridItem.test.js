import React from 'react';
import { GifGridItem } from '../../components/GifGridItem';
import { shallow } from 'enzyme';

describe('GifGridItem.js tests', () => {
    const title = 'Perro chocolo';
    const url = 'https://localhost/jefe.jpg';
    const wrapper = shallow(<GifGridItem title={title} url={url} />);

    test('Shoul show the GifGridItem component', () => {
        expect(wrapper).toMatchSnapshot();
    });

    test('The title should be present in the component', () => {
        const p = wrapper.find('p');
        expect(p.text().trim()).toBe(title);
    });

    test('Test for src and alt in the img', () => {
        const img = wrapper.find('img');
        const src = img.props().src;
        const alt = img.props().alt;
        expect(src).toBe(url);
        expect(alt).toBe(title);
    });

    test('Class css test', () => {
        const div = wrapper.find('div');
        const hasClass = div.hasClass('animate__bounceIn');
        expect(hasClass).toBe(true);
    });

});
