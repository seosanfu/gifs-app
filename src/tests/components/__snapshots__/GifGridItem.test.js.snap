// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`GifGridItem.js tests Shoul show the GifGridItem component 1`] = `
<div
  className="card animate__animated animate__bounceIn"
>
  <img
    alt="Perro chocolo"
    src="https://localhost/jefe.jpg"
  />
  <p>
    Perro chocolo
  </p>
</div>
`;
